lib/abi/function_selector.ex

```diff
@@ -415,6 +415,18 @@ defmodule ABI.FunctionSelector do
     end
   end
 
+  defp add_event_id(selector) do
+    signature = encode(selector)
+
+    case ExKeccak.hash_256(signature) do
+      <<event_id::binary-size(32)>> ->
+        %{selector | method_id: event_id}
+
+      _ ->
+        selector
+    end
```

Comment:
Since no extra extraction logic is required here I would say it can be even simpler.

```suggestion
      %{selector | method_id: ExKeccak.hash_256(signature)}
    end
```